microsoft/PowerToys
# [FancyZones] Open window in a zone fix

## Summary of the Pull Request

**What is this about:**

In some cases, app zone history is not found due to a virtual desktop id, so the window is not snapped to a zone after opening.
Note: this doesn't fix the problem with FileExplorer.  

**What is included in the PR:** 

* Centralized work with virtual desktops in one place.
* Changed virtual desktop comparison in the app-zone-history. In some cases, windows weren't snapped to a zone due to virtual desktop ids.  
* Added separate function for retrieving process path waiting for the uwp app to be opened to get the actual app path instead of ApplicationFrameHost.
* Added logs 

**How does someone test / validate:** 

In my case, I can reproduce a bug with a non-zoned window with Skype. 
Allow Skype to run on startup and uncheck `Settings -> General -> Launch Skype in the background`.

* Remove virtual desktops from the registry
* Snap window to a zone. 
* Restart
* Verify that the app is opened in a zone
* Create a new virtual desktop
* Restart
* Verify that the app is opened in a zone

## Quality Checklist

- [x] **Linked issue:** #https://github.com/microsoft/PowerToys/issues/17027, https://github.com/microsoft/PowerToys/issues/18035, https://github.com/microsoft/PowerToys/issues/17550, https://github.com/microsoft/PowerToys/issues/13759, https://github.com/microsoft/PowerToys/issues/17909

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
